framework/src/modules/dpos/dpos_module.ts

```diff
@@ -12,90 +12,223 @@
-							dataType: 'uint64',
-							fieldNumber: 2,
-						},
-					},
-					required: ['delegateAddress', 'amount'],
-				},
-			},
-			unlocking: {
-				type: 'array',
-				fieldNumber: 3,
-				items: {
-					type: 'object',
-					properties: {
-						delegateAddress: {
-							dataType: 'bytes',
-							fieldNumber: 1,
-						},
-						amount: {
-							dataType: 'uint64',
-							fieldNumber: 2,
-						},
-						unvoteHeight: {
-							dataType: 'uint32',
-							fieldNumber: 3,
-						},
-					},
-					required: ['delegateAddress', 'amount', 'unvoteHeight'],
-				},
-			},
-		},
-		default: {
-			delegate: {
-				username: '',
-				pomHeights: [],
-				consecutiveMissedBlocks: 0,
-				lastForgedHeight: 0,
-				isBanned: false,
-				totalVotesReceived: BigInt(0),
-			},
-			sentVotes: [],
-			unlocking: [],
-		},
-	};
+	public accountSchema = dposAccountSchema;
+
+	public readonly rounds: Rounds;
+
+	private readonly _activeDelegates: number;
+	private readonly _standbyDelegates: number;
+	private readonly _delegateListRoundOffset: number;
+	private readonly _delegatesPerRound: number;
+	private readonly _delegateActiveRoundLimit: number;
+	private readonly _blockTime: number;
+	private _finalizedHeight = 0;
+
+	public constructor(config: GenesisConfig) {
+		super(config);
+
+		const errors = validator.validate(dposModuleParamsSchema, this.config);
+		if (errors.length) {
+			throw new LiskValidationError([...errors]);
+		}
+
+		this._activeDelegates = this.config.activeDelegates as number;
+		this._standbyDelegates = this.config.standbyDelegates as number;
+		this._delegateListRoundOffset = this.config.delegateListRoundOffset as number;
+		this._delegatesPerRound = this._activeDelegates + this._standbyDelegates;
+		this._blockTime = config.blockTime;
+		this._delegateActiveRoundLimit = 3;
+
+		this.rounds = new Rounds({ blocksPerRound: this._delegatesPerRound });
+	}
+
+	// eslint-disable-next-line consistent-return
+	public async afterBlockApply(input: AfterBlockApplyInput): Promise<void> {
+		const finalizedHeight = input.consensus.getFinalizedHeight();
+		const lastBootstrapHeight = input.consensus.getLastBootstrapHeight();
+		const isLastBlockOfRound = this._isLastBlockOfTheRound(input.block.header);
+		const { height } = input.block.header;
+		const isBootstrapPeriod = height <= lastBootstrapHeight;
+
+		if (finalizedHeight !== this._finalizedHeight) {
+			this._finalizedHeight = finalizedHeight;
+
+			const finalizedBlockRound = this.rounds.calcRound(finalizedHeight);
+			const disposableDelegateListUntilRound =
+				finalizedBlockRound - this._delegateListRoundOffset - this._delegateActiveRoundLimit;
+			await deleteVoteWeightsUntilRound(disposableDelegateListUntilRound, input.stateStore);
+		}
+
+		if (!isBootstrapPeriod) {
+			// Calculate account.dpos.delegate.consecutiveMissedBlocks and account.dpos.delegate.isBanned
+			await this._updateProductivity(input);
+		}
+
+		if (!isLastBlockOfRound) {
+			return Promise.resolve();
```

Comment:
I think u dont need to return Promise.resolve since its async function

Reply:
it was some IDE linting issue which was complaining to return Promise.